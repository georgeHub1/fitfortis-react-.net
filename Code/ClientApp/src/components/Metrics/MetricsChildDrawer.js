import React, { Component } from 'react';
import { Button, InputNumber, Select, Icon, Switch } from 'antd';
import { FormattedMessage, injectIntl } from 'react-intl';
import styles from './styles.module.less';
import { colors, DEFAULT_STROKE } from '../../constants/metrics';
import PropTypes from 'prop-types';

const Option = Select.Option;

class MetricsChildDrawer extends Component {
  displayName = MetricsChildDrawer.name;

  static propTypes = {
    showGoalLines: PropTypes.bool.isRequired,
    annotateLastEntry: PropTypes.bool.isRequired,
    annotateMaxEntry: PropTypes.bool.isRequired,
    annotateMinEntry: PropTypes.bool.isRequired,
    stroke: PropTypes.string.isRequired,
    isUpdating: PropTypes.bool.isRequired,
    close: PropTypes.func.isRequired,
    changeProperties: PropTypes.func.isRequired,
    type: PropTypes.number.isRequired,
    yMin:  PropTypes.number,
    yMax:  PropTypes.number,
    goal: PropTypes.number,
    defaultGoalMin: PropTypes.number,
    defaultGoalMax: PropTypes.number,
    isGoalValueEditable: PropTypes.bool,
    metricId: PropTypes.string
  }
  constructor (props) {
    super(props);
    this.state = {
      showGoalLines: this.props.showGoalLines,
      annotateLastEntry: this.props.annotateLastEntry,
      annotateMaxEntry: this.props.annotateMaxEntry,
      annotateMinEntry: this.props.annotateMinEntry,
      stroke: this.props.stroke || DEFAULT_STROKE,
      yMin:  this.props.yMin,
      yMax:  this.props.yMax,
      goal: this.props.goal,
      GoalMin: this.props.defaultGoalMin,
      GoalMax: this.props.defaultGoalMax
    };
  }

  handleColor = color => {
    this.setState({ stroke: color });
  };

  handleApply = () => {
    const changedValues = Object.keys(this.state)
      .filter(key => this.state[key] !== this.props[key])
      .length;

    if (changedValues === 0) {
      return this.props.close();
    }

    this.props.changeProperties(this.state);
  };

  handleGoal = e => this.setState({ goal: e });

  handleGoalMin = e => this.setState({ GoalMin: e });

  handleGoalMax = e => this.setState({ GoalMax: e });

  handleMinRange = e => this.setState({ yMin: e });

  handleMaxRange = e => this.setState({ yMax: e });
  onChangeSwitch = (value, formatKey, key) => {
    this.setState(state => ({
      ...state,
      [key]: value
    }));
  };

  getToogle = state => {
    return [
      {
        toggle: state.showGoalLines,
        formatKey: 'toggleGL',
        key: 'showGoalLines'
      },
      {
        toggle: state.annotateLastEntry,
        formatKey: 'toggleLast',
        key: 'annotateLastEntry'
      },
      {
        toggle: state.annotateMaxEntry,
        formatKey: 'toggleMax',
        key: 'annotateMaxEntry'
      },
      {
        toggle: state.annotateMinEntry,
        formatKey: 'toggleMin',
        key: 'annotateMinEntry'
      }
    ];
  }

  render () {
    const {
      stroke,
      yMin,
      yMax,
      goal,
      GoalMin,
      GoalMax
     } = this.state;
    const { close, isUpdating, units, type : metricType, intl: { formatMessage } } = this.props;
    const toggles = this.getToogle(this.state);

    return (
      <div>
       <div className="color_mertic_setting">
        <div className={styles.itemSetting}>
          <p className={styles.nameOfSetting}>
            <FormattedMessage id="MetricsChildDrawer.color" />
          </p>
          <Select
            className="colorPickerSelect"
            style={{ background: stroke, width: 200, borderRadius: 14 }}
            defaultValue={stroke}
            dropdownClassName="colorPickerDropdown"
            onChange={this.handleColor}
          >
            {colors.map(color => (
              <Option
                style={{ background: color, color: '#fff' }}
                key={color}
                value={color}
              >
                {stroke === color && <Icon type="check" />}
              </Option>
            ))}
          </Select>
        </div>
        {(metricType === 1) && (
          <div>
            <div className={styles.itemSetting}>
              <p className={styles.nameOfSetting}>
                <FormattedMessage id="MetricsChildDrawer.goal" /> ({units})
              </p>
              <InputNumber
                className={styles.InputNumber}
                placeholder={formatMessage({
                  id: 'MetricsChildDrawer.goalPlaceholder'
                })}
                value={goal}
                onChange={this.handleGoal}
              />
            </div>
          </div>
        )}
        {(metricType === 2) && (
          <div>
            {
              <div className={styles.itemSetting}>
                <p className={styles.nameOfSetting}>
                  <FormattedMessage id="MetricsChildDrawer.goalMax" /> ({units})
                </p>
                <InputNumber
                  className={styles.InputNumber}
                  placeholder={formatMessage({
                    id: 'MetricsChildDrawer.goalPlaceholder'
                  })}
                  value={GoalMax}
                  onChange={this.handleGoalMax}
                />
              </div>
            }
            {
              <div className={styles.itemSetting}>
                <p className={styles.nameOfSetting}>
                  <FormattedMessage id="MetricsChildDrawer.goalMin" /> ({units})
                </p>
                <InputNumber
                  className={styles.InputNumber}
                  placeholder={formatMessage({
                    id: 'MetricsChildDrawer.goalPlaceholder'
                  })}
                  value={GoalMin}
                  onChange={this.handleGoalMin}
                />
              </div>
            }
          </div>
        )}

        {(metricType === 3) && (
          <div>
            <div className={styles.itemSetting}>
              <p className={styles.nameOfSetting}>
                <FormattedMessage id="MetricsChildDrawer.rangeMax" /> ({units})
              </p>
              <InputNumber
                className={styles.InputNumber}
                placeholder={formatMessage({
                  id: 'MetricsChildDrawer.rangePlaceholder'
                })}
                value={yMax}
                onChange={this.handleMaxRange}
              />
            </div>

            <div className={styles.itemSetting}>
              <p className={styles.nameOfSetting}>
                <FormattedMessage id="MetricsChildDrawer.rangeMin" /> ({units})
              </p>
              <InputNumber
                className={styles.InputNumber}
                placeholder={formatMessage({
                  id: 'MetricsChildDrawer.rangePlaceholder'
                })}
                value={yMin}
                onChange={this.handleMinRange}
              />
            </div>
          </div>
        )}
        {toggles.map((el, i) => (
          <div key={el.formatKey} className={styles.itemSetting}>
            <p className={styles.nameOfSetting}>
              <FormattedMessage id={`MetricsChildDrawer.${el.formatKey}`} />
            </p>
            <Switch
              checked={el.toggle}
              onChange={value => this.onChangeSwitch(value, el.formatKey, el.key)}
            />
          </div>
        ))}

      </div >
        <div className={styles.btnBlock}>
          <Button aria-label="close button" onClick={close} >
            <FormattedMessage id="MetricsChildDrawer.close" />
          </Button>
          <Button
            type="primary"
            aria-label="apply button"
            onClick={this.handleApply}
          >
            {isUpdating ? <Icon type="loading" />
              : <FormattedMessage id="MetricsChildDrawer.apply" />
            }
          </Button>
        </div>
        </div>
    );
  }
}

export default injectIntl(MetricsChildDrawer);
