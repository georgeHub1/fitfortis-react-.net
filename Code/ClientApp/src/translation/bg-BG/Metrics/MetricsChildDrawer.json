{
  "close": "Затвори",
  "apply": "Запиши",
  "color": "Цвят",
  "goalPlaceholder": "стойност",
  "goal": "Цел",
  "goalMin": "Цел (Минимум)",
  "goalMax": "Цел (Максимум)",
  "rangeMin": "Минимум",
  "rangeMax": "Максимум",
  "rangePlaceholder": "стойност",
  "toggleGL": "Покажи целта върху графиката",
  "toggleMin": "Анотирай минималната стойност",
  "toggleMax": "Анотирай максималната стойност",
  "toggleLast": "Анотирай последната стойност"
}
