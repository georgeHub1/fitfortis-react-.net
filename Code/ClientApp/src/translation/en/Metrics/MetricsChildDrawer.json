{
  "close": "Close",
  "apply": "Apply",
  "color": "Color",
  "goalPlaceholder": "value",
  "goal": "Goal",
  "goalMin": "Goal (Min)",
  "goalMax": "Goal (Max)",
  "rangeMin": "Min",
  "rangeMax": "Max",
  "rangePlaceholder": "value",
  "toggleGL": "Show goal lines",
  "toggleMin": "Annotate min entry",
  "toggleMax": "Annotate max entry",
  "toggleLast": "Annotate latest entry"
}
