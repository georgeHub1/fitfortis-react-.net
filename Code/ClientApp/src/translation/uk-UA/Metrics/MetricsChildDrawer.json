{
  "close": "Закрити",
  "apply": "Застосувати",
  "color": "Колір",
  "goalPlaceholder": "число",
  "goal": "Ціль",
  "goalMin": "Ціль (Мінімальний діапазон)",
  "goalMax": "Ціль (Максимальний діапазон)",
  "rangeMin": "Мінімальний діапазон",
  "rangeMax": "Максимальний діапазон",
  "rangePlaceholder": "число",
  "toggleGL": "Показати лінії цілі",
  "toggleMin": "Аннотація мінімального значення",
  "toggleMax": "Аннотація максимального значення",
  "toggleLast": "Аннотація останнього значення"
}
