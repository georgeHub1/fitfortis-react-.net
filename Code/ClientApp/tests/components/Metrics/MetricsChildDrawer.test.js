import React from 'react';
import renderer from 'react-test-renderer';
import MetricsChildDrawer  from '../../../src/components/Metrics/MetricsChildDrawer.js';

const Container = MetricsChildDrawer;

describe('MetricsChildDrawer', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<Container
      showGoalLines={false}
      annotateLastEntry={false}
      annotateMaxEntry={false}
      annotateMinEntry={false}
      stroke={''}
      isUpdating={false}
      close={jest.fn()}
      changeProperties={jest.fn()}
      type={0}
    />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
