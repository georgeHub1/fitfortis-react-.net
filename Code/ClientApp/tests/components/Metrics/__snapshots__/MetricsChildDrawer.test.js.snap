// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricsChildDrawer renders correctly 1`] = `
<div>
  <div
    className="color_mertic_setting"
  >
    <div>
      <p />
      <div
        className="colorPickerSelect ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "#335889",
            "borderRadius": 14,
            "width": 200,
          }
        }
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="#335889"
            >
              #335889
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </div>
    <div>
      <p />
      <button
        aria-checked={false}
        checked={false}
        className="ant-switch"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        role="switch"
        type="button"
      >
        <span
          className="ant-switch-inner"
        />
      </button>
    </div>
    <div>
      <p />
      <button
        aria-checked={false}
        checked={false}
        className="ant-switch"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        role="switch"
        type="button"
      >
        <span
          className="ant-switch-inner"
        />
      </button>
    </div>
    <div>
      <p />
      <button
        aria-checked={false}
        checked={false}
        className="ant-switch"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        role="switch"
        type="button"
      >
        <span
          className="ant-switch-inner"
        />
      </button>
    </div>
    <div>
      <p />
      <button
        aria-checked={false}
        checked={false}
        className="ant-switch"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        role="switch"
        type="button"
      >
        <span
          className="ant-switch-inner"
        />
      </button>
    </div>
  </div>
  <div>
    <button
      aria-label="close button"
      className="ant-btn"
      onClick={[Function]}
      type="button"
    />
    <button
      aria-label="apply button"
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
